example/spec/ingestGranule/IngestUMMGSuccessSpec.js

```diff
@@ -427,21 +419,28 @@ describe('The S3 Ingest Granules workflow configured to ingest UMM-G', () => {
     it('updates the UMM-G JSON file in S3 with new paths', async () => {
       const updatedUmm = await getUmmObject(newS3UMMJsonFileLocation);
 
-      const relatedUrlDifferences = updatedUmm.RelatedUrls.filter((urlObject) => {
-        // Skip non-science URLs and public S3 URLs
-        if (!isUMMGScienceUrl(urlObject.URL) ||
-            urlObject.URL.match(/s3\.amazonaws\.com/)) {
-          return false;
-        }
-        const relatedUrl = new URL(urlObject.URL);
-        relatedUrl.host = process.env.DISTRIBUTION_ENDPOINT;
-        return !originalUmmUrls.includes(relatedUrl.toString());
-      });
+      const changedUrls = updatedUmm.RelatedUrls
+        .filter((urlObject) => urlObject.URL.match(/.*.hdf$/))
+        .map((urlObject) => urlObject.URL);
+      const unchangedUrls = updatedUmm.RelatedUrls
+        .filter((urlObject) => !urlObject.URL.match(/.*.hdf$/))
+        .map((urlObject) => urlObject.URL);
 
       // Only the file that was moved was updated
-      expect(relatedUrlDifferences.length).toEqual(1);
-
-      expect(relatedUrlDifferences[0].URL).toContain(destinationKey);
+      expect(changedUrls.length).toEqual(1);
+      expect(changedUrls[0]).toContain(destinationKey);
+
+      const unchangedOriginalUrls = originalUmmUrls.filter((original) => !original.match(/.*.hdf$/));
+      expect(unchangedOriginalUrls.length).toEqual(unchangedUrls.length);
+
+      // Each originalUmmUrl (removing the DISTRIBUTION_ENDPOINT) should be found
+      // in one of the updated URLs. We have to do this comparison because the
+      // setup tests uses a fake endpoint, but it's possible that the api has
+      // the actual endpoint.
+      unchangedOriginalUrls.forEach((original) => {
+        const base = original.replace(process.env.DISTRIBUTION_ENDPOINT, '');
```

Comment:
I kinda prefer using `URL` to replace the host and testing whether the whole URL exists via `unchangedUrls.includes(url)` instead of testing against just the URL path, but it seems like you're saying that wasn't a valid test?

Reply:
Ok, I see where you are asking.  So here's a concrete example.

The originalUmmUrls are:

```
[ 'https://nasa.github.io/cumulus/docs/cumulus-docs-readme',
  'http://localhost:5002/mhs3-protected/MOD09GQ___006/2016/MOD/MOD09GQ.A8917090.uSYPQX.006.5582417209822.hdf',
  'http://localhost:5002/mhs3-public/MOD09GQ___006/MOD/MOD09GQ.A8917090.uSYPQX.006.5582417209822_ndvi.jpg',
  'http://localhost:5002/mhs3-protected-2/MOD09GQ___006/MOD/MOD09GQ.A8917090.uSYPQX.006.5582417209822.cmr.json',
  'http://localhost:5002/s3credentials' ]
```

And the updated ones when ApiEndpoints has a valid DISTRIBUTION_API:

```
[ 'https://nasa.github.io/cumulus/docs/cumulus-docs-readme',
  'https://rz0l1ei1p1.execute-api.us-east-1.amazonaws.com/dev/mhs3-public/MOD09GQ___006/MOD/MOD09GQ.A8917090.uSYPQX.006.5582417209822_ndvi.jpg',
  'https://rz0l1ei1p1.execute-api.us-east-1.amazonaws.com/dev/mhs3-protected-2/MOD09GQ___006/MOD/MOD09GQ.A8917090.uSYPQX.006.5582417209822.cmr.json',
  'https://rz0l1ei1p1.execute-api.us-east-1.amazonaws.com/dev/mhs3-protected/mhs3-IngestUMMGSuccess-1554911678203-test-data/files/MOD09GQ___006/2016/MOD/MOD09GQ.A8917090.uSYPQX.006.5582417209822.hdf',
  'https://rz0l1ei1p1.execute-api.us-east-1.amazonaws.com/dev/s3credentials' ]
```

if you replace host with process.env.DISTRIBUTION_API

```
[ 'https://nasa.github.io/cumulus/docs/cumulus-docs-readme',
  'https://localhost:5002/dev/mhs3-public/MOD09GQ___006/MOD/MOD09GQ.A8917090.uSYPQX.006.5582417209822_ndvi.jpg',
  'https://localhost:5002/dev/mhs3-protected-2/MOD09GQ___006/MOD/MOD09GQ.A8917090.uSYPQX.006.5582417209822.cmr.json',
  'https://localhost:5002/dev/mhs3-protected/mhs3-IngestUMMGSuccess-1554911678203-test-data/files/MOD09GQ___006/2016/MOD/MOD09GQ.A8917090.uSYPQX.006.5582417209822.hdf',
  'https://localhost:5002/dev/s3credentials' ]
```

They won't match.